SDK-Onboarding/readme.md

```diff
@@ -0,0 +1,243 @@
+
+#### Generation Directives
+When generating client, we sometimes need to do additional customizations ranging from fixing swagger and generation errors, renaming methods or models, to deleting unwanted or redundant apis.
+Read articles [here](https://github.com/Azure/autorest/blob/main/docs/generate/directives.md), and [here](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md).
+
+#### Generated Code Customization (_patch file)
+In order to further customize your client, you can actually override generated code using special _patch file.
+Read article [here](https://github.com/Azure/autorest.python/blob/autorestv3/docs/customizations.md).  
+
+## Samples
+At this stage, we write samples to showcase how to use the SDK.
+For preview releases, we mainly write samples for hero-scenarios.
+You need to provide sync and async samples.
+You also need to provide a readme file explaining all your samples.
+And you need to make sure everything runs without errors, otherwise, the CI pipeline won't pass on your PR.
+
+## Testing
+At this stage, we write tests to make sure our client and methods are working correctly. They are mainly E2E tests, calling client methods and checking for valid response and data. 
+
+### Testing Mode
+You can run tests in 2 modes: live and recorded.
+In live mode, your client actually makes calls to the service endpoints. But in recorded mode, the dev env simply mocks network traffic.
+
+### Recording Tests
+When writing tests, you need to modify local environment to run tests in live mode, and after doing this, you'll see recoded tests as many file.yml added containing network info for each call. At this point, you can run tests in recoded mode.
+
+Please check your [language-specific section](#language-specific-instructions-1) for how to enable each.  
+
+
+## Final Touches
+At this stage, you need to privde final touches
+
+### Changelog.md
+Update changelog.md file to add entry for the currnt release
+with things like version, date, new features, and breaking changes (if any).
+
+### Readme.md
+This will be the entrypoint for all developers looking for your SDK. You should add things like an overview of what the service does and some samples.
+
+## SDK Release
+At this stage we trigger release pipeline on Azure DevOps to release a package for users (pip package for python, or nuget package for .net).
+
+Check with senior people from SDK team (because you most likely won't have permissions to carry out this step).
+
+## CI/CD Pipeline
+
+### Testing pipeline
+We simply have recorded mode because when submitting a PR or pushing new changes, CI/CD pipeline will only run tests in recorded mode to reduce network traffic. But it also has nightly builds which run tests in live mode. 
+
+### Secrets management
+Every project has a tests.yml file which contains info you need for testing like env variables names used by pipeline.
```

Comment:
We use tests.yml if we have static env vars, but it's preferred (if possible) to dynamically create the env vars in the test-resources.json and use those in our tests. 

https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/51/Testing-Guidelines